ddtrace/internal/gitmetadata.py

```diff
@@ -0,0 +1,112 @@
+import os
+import typing
+
+from ddtrace.internal.utils import formats
+
+
+__GITMETADATA_TAGS = None  # type: typing.Optional[typing.Dict[str, str]]
+
+ENV_ENABLED_FLAG = "DD_TRACE_GIT_METADATA_ENABLED"
+ENV_REPOSITORY_URL = "DD_GIT_REPOSITORY_URL"
+ENV_COMMIT_SHA = "DD_GIT_COMMIT_SHA"
+ENV_MAIN_PACKAGE = "DD_MAIN_PACKAGE"
+ENV_GLOBAL_TAGS = "DD_TAGS"
+
+TAG_REPOSITORY_URL = "git.repository_url"
+TAG_COMMIT_SHA = "git.commit.sha"
+
+TRACE_TAG_REPOSITORY_URL = "_dd.git.repository_url"
+TRACE_TAG_COMMIT_SHA = "_dd.git.commit.sha"
+
+
+def __get_tags_from_env():
```

Comment:
We already have [`envier`](https://github.com/datadog/envier) as a dependency in `ddtrace`. I wonder if we could use that to fetch from the environment and avoid custom caching logic here.

Reply:
Hmm, interesting, never know about this module. Thanks!

But here, I used the same way, as [tracer ](https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/settings/config.py#L199) and this seems to be more convenient because we need custom logic to parse tags.

Reply:
Envier variable declarations allow for a custom parser, so it should be flexible enough. Our long-term plan is to eventually replace every direct env access with envier, so if we could avoid any direct env accesses going forward it will make our life easier later 🙂.

Reply:
Done!